pkg/controller/sparkapplication/controller.go

```diff
@@ -277,6 +258,8 @@ func (c *Controller) processNextItem() bool {
 	}
 	defer c.queue.Done(key)
 
+	glog.V(2).Infof("Starting processing key: %v", key)
```

Comment:
oldApp.ResourceVersion == newApp.ResourceVersion
  is a dangerous check. We will never make progress if we get error in writing

I would rather check terminal state

Reply:
Also lets handle the retries in the state machine.